docs/content/drafts/DropdownMenu2.mdx

```diff
@@ -0,0 +1,308 @@
+  const anchorRef = React.createRef()
+
+  return (
+    <>
+      <Button ref={anchorRef} onClick={() => setOpen(!open)}>
+        {open ? 'Close Menu' : 'Open Menu'}
+      </Button>
+
+      <DropdownMenu open={open} onOpenChange={setOpen} anchorRef={anchorRef}>
+        <DropdownMenu.Overlay>
+          <ActionList>
+            <ActionList.Item selected={true}>Text</ActionList.Item>
+            <ActionList.Item>Number</ActionList.Item>
+            <ActionList.Item>Date</ActionList.Item>
+            <ActionList.Item>Iteration</ActionList.Item>
+          </ActionList>
+        </DropdownMenu.Overlay>
+      </DropdownMenu>
+    </>
+  )
+}
+
+render(<Example />)
+```
+
+### With Overlay Props
+
+```javascript live noinline
+// import {DropdownMenu, ActionList} from '@primer/react/drafts'
+const {DropdownMenu, ActionList} = drafts // ignore docs silliness; import like that ↑
+
+const fieldTypes = [
+  {icon: TypographyIcon, name: 'Text'},
+  {icon: NumberIcon, name: 'Number'},
+  {icon: CalendarIcon, name: 'Date'},
+  {icon: SingleSelectIcon, name: 'Single select'},
+  {icon: IterationsIcon, name: 'Iteration'}
+]
+
+const Example = () => {
+  const handleEscape = () => alert('you hit escape!')
+
+  const [selectedIndex, setSelectedIndex] = React.useState(1)
+  const selectedType = fieldTypes[selectedIndex]
+
+  return (
+    <DropdownMenu>
+      <DropdownMenu.Button aria-label="Select field type" leadingIcon={selectedType.icon}>
+        {selectedType.name}
+      </DropdownMenu.Button>
+      <DropdownMenu.Overlay width="medium" onEscape={handleEscape}>
+        <ActionList>
+          {fieldTypes.map((type, index) => (
+            <ActionList.Item key={index} selected={index === selectedIndex} onSelect={() => setSelectedIndex(index)}>
+              <type.icon /> {type.name}
+            </ActionList.Item>
+          ))}
+        </ActionList>
+      </DropdownMenu.Overlay>
+    </DropdownMenu>
+  )
+}
+
+render(<Example />)
+```
+
+### With a custom anchor
+
+You can choose to have a different _anchor_ for the Menu dependending on the application's context.
+
+&nbsp;
+
+```javascript live noinline
+// import {DropdownMenu, ActionList} from '@primer/react/drafts'
+const {DropdownMenu, ActionList} = drafts // ignore docs silliness; import like that ↑
+
+render(
+  <DropdownMenu>
+    <DropdownMenu.Anchor>
+      <button>Select a field type</button>
+    </DropdownMenu.Anchor>
+
+    <DropdownMenu.Overlay>
+      <ActionList>
+        <ActionList.Item selected={true}>Text</ActionList.Item>
+        <ActionList.Item>Number</ActionList.Item>
+        <ActionList.Item>Date</ActionList.Item>
+        <ActionList.Item>Iteration</ActionList.Item>
+      </ActionList>
+    </DropdownMenu.Overlay>
+  </DropdownMenu>
+)
+```
+
+<Note variant="warning">
+
+Use `DropdownMenu` to select an option from a small list. If you’re looking for filters or multiple selection, use [SelectPanel](/SelectPanel) instead.
```

Comment:
This feels quite important, shall we hoist it near the top of page so it has better visibility? We've done this for other components before.

Reply:
This came up before with ActionMenu. What we decided was that [interface guidelines](https://primer.style/design/components/action-list#application) is the first place where you learn which component to use, and this is the second note near `props` just in case you missed that.